### ISSUE TITLE
Json and friends

### ISSUE THREAD

**Issue by Morningstar88** [Morningstar88/kalki-search#88]
Yes, whitespace can be significant in JSON in some situations, but not always. Here's a breakdown:

Where Whitespace Matters:

Within Strings: Whitespace characters inside JSON strings are treated like any other character and are preserved. For example:
JSON
{ "key1": "This is a string with spaces" } 
Use code [with caution.](https://gemini.google.com/faq#coding)
Around Property Names: Whitespace around property names is required. For example, the following is invalid JSON:
JSON
{"key1" :"value1"} 
Use code [with caution.](https://gemini.google.com/faq#coding)
The correct format would be:
JSON
{ "key1" : "value1" }
Use code [with caution.](https://gemini.google.com/faq#coding)
Where Whitespace Doesn't Matter:

Between Structural Elements: You can freely add extra spaces, tabs, or newlines between commas, colons, brackets, and braces to improve readability without affecting the meaning of your JSON data. For example, the following JSON variations are functionally equivalent:
JSON
{"key1": "value1", "key2": "value2"}

{ 
  "key1" : "value1",
  "key2" : "value2" 
}
Use code [with caution.](https://gemini.google.com/faq#coding)
Potential Issues:

Unexpected Characters: If there's an unexpected non-whitespace character after your JSON data (like in a misformatted file), you might encounter an error like "SyntaxError: JSON.parse: unexpected non-whitespace character after JSON data".
Special Serializer Behavior: Some JSON serializers (like .NET's DataContractJsonSerializer) might have specific quirks regarding whitespace, potentially requiring it in unexpected places.
In Summary:  While whitespace is often used for formatting and readability in JSON, it's crucial to remember that it can be significant within strings and around property names.

**Comment by Morningstar88** [comment]
I'd be glad to assist you with finding suitable JSON linters. Here are some of the most recommended options, incorporating insights from expert ratings and addressing potential issues:

Command-Line Tools:

JSONLint:
Pros: Versatile, supports minified JSON, validates against RFC standards, configurable output format.
Cons: Can be opinionated in its validation (e.g., disallowing trailing commas).
